### PR TITLE
Unittests

### PR DESCRIPTION
**Pull request by artemgael96**

### CODE REVIEW

File: InternshipBE/ExaLearn.Tests/Controllers/AuthenticateControllerTests.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    {
+        private readonly Mock<UserManager<User>> _mockUserManager;
+        private readonly Mock<IConfiguration> _mockConfiguration;
+
+        public AuthorizationControllerTests()
+        {
+            var store = new Mock<IUserStore<User>>();
+
+            _mockUserManager = new Mock<UserManager<User>>(store.Object, null, null, null, null, null, null, null, null);
+            _mockUserManager.Setup(u => u.FindByNameAsync(It.IsAny<string>())).Returns(AuthenticateControllerrFixture.TestFindByNameAsync());
```
Comment (romanstat) [inline InternshipBE/ExaLearn.Tests/Controllers/AuthenticateControllerTests.cs]: GetUserNameAsync

File: InternshipBE/ExaLearn.Tests/Controllers/AuthenticateControllerTests.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            _mockUserManager.Setup(u => u.CheckPasswordAsync(It.IsAny<User>(), It.IsAny<string>())).Returns(AuthenticateControllerrFixture.TestCheckPasswordAsyncTrue());
```
Comment (romanstat) [inline InternshipBE/ExaLearn.Tests/Controllers/AuthenticateControllerTests.cs]: IsCorrectPasswordAsync

File: InternshipBE/ExaLearn.Tests/Controllers/AuthenticateControllerTests.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        {
+            // Arrange
+            var _controller = new AuthenticateController(_mockUserManager.Object, _mockConfiguration.Object);
+            var loginModel = new LoginModel();
+
+            // Act
+            dynamic result = await _controller.Login(loginModel);
+
+            // Assert
+            Assert.NotNull(result);
```
Comment (romanstat) [inline InternshipBE/ExaLearn.Tests/Controllers/AuthenticateControllerTests.cs]: add checks

File: InternshipBE/ExaLearn.Tests/Fixture/AuthenticateControllerrFixture.cs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+namespace ExaLearn.Tests.Fixture
+{
+    public static class AuthenticateControllerrFixture
+    {
+        public static async Task<User> TestFindByNameAsync()
+        {
+            return new User() { UserName = "adminexa@mailnesia.com" };
+        }
+
+        public static async Task<bool> TestCheckPasswordAsyncTrue()
```
Comment (romanstat) [inline InternshipBE/ExaLearn.Tests/Fixture/AuthenticateControllerrFixture.cs]: rename

File: InternshipBE/ExaLearn.Tests/Fixture/AuthenticateControllerrFixture.cs [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        {
+            return true;
+        }
+
+        public static async Task<bool> TestCheckPasswordAsyncFalse()
```
Comment (romanstat) [inline InternshipBE/ExaLearn.Tests/Fixture/AuthenticateControllerrFixture.cs]: concat

File: InternshipBE/ExaLearn.Tests/Controllers/AuthenticateControllerTests.cs (hunk 3 above)
Comment (romanstat) [reply InternshipBE/ExaLearn.Tests/Controllers/AuthenticateControllerTests.cs]: experation

Comment (romanstat) [reply InternshipBE/ExaLearn.Tests/Controllers/AuthenticateControllerTests.cs]: token structure with regular expression

File: InternshipBE/ExaLearn.Tests/Fixture/AuthenticateControllerrFixture.cs [hunk 6]
```diff
@@ -0,0 +1,24 @@
+﻿using ExaLearn.Dal.Entities;
+using System.Threading.Tasks;
+
+namespace ExaLearn.Tests.Fixture
+{
+    public static class AuthenticateControllerrFixture
+    {
+        public static async Task<User> GetUserNameAsync()
```
Comment (romanstat) [inline InternshipBE/ExaLearn.Tests/Fixture/AuthenticateControllerrFixture.cs]: =>

File: InternshipBE/ExaLearn.Tests/Fixture/AuthenticateControllerrFixture.cs [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+        {
+            return new User() { UserName = "adminexa@mailnesia.com" };
+        }
+
+        public static async Task<bool> IsCorrectPasswordAsync()
```
Comment (romanstat) [inline InternshipBE/ExaLearn.Tests/Fixture/AuthenticateControllerrFixture.cs]: =>

**Review by romanstat — APPROVED** [review]

**Comment by artemgael96** [comment]
failed merge confilct

**Pull request closed without merging**